pkg/cmd/server/origin/run_components.go

```diff
@@ -265,9 +288,15 @@ func (c *MasterConfig) RunDeploymentController() {
 	if err != nil {
 		glog.Fatalf("Unable to initialize deployment controller: %v", err)
 	}
+
+	port, err := kubeserver.GetKubernetesMasterPort(&c.Options)
+	if err != nil {
+		glog.Fatal(err)
+	}
+	master := fmt.Sprintf("https://kubernetes.default.svc.%s:%d", getDNSDomain(&c.Options, false), port)
 	// TODO eliminate these environment variables once service accounts provide a kubeconfig that includes all of this info
 	env := clientcmd.EnvVars(
-		kclientConfig.Host,
+		master,
```

Comment:
I don't think this will work... the master shouldn't talk to itself on dns to find itself